e2e/rbd.go

```diff
@@ -23,47 +23,88 @@ var (
 	rbdExamplePath     = "../examples/rbd/"
 	rbdDeploymentName  = "csi-rbdplugin-provisioner"
 	rbdDaemonsetName   = "csi-rbdplugin"
-	namespace          = "default"
 )
 
 func deployRBDPlugin() {
 	// delete objects deployed by rook
-	framework.RunKubectlOrDie("delete", "--ignore-not-found=true", "-f", rbdDirPath+rbdProvisionerRBAC)
-	framework.RunKubectlOrDie("delete", "--ignore-not-found=true", "-f", rbdDirPath+rbdNodePluginRBAC)
-	// deploy provisioner
-	framework.RunKubectlOrDie("create", "-f", rbdDirPath+rbdProvisioner)
-	framework.RunKubectlOrDie("create", "-f", rbdDirPath+rbdProvisionerRBAC)
-	framework.RunKubectlOrDie("create", "-f", rbdDirPath+rbdProvisionerPSP)
-	// deploy nodeplugin
-	framework.RunKubectlOrDie("create", "-f", rbdDirPath+rbdNodePlugin)
-	framework.RunKubectlOrDie("create", "-f", rbdDirPath+rbdNodePluginRBAC)
-	framework.RunKubectlOrDie("create", "-f", rbdDirPath+rbdNodePluginPSP)
+	data, err := replaceNamespaceInTemplate(rbdDirPath + rbdProvisionerRBAC)
+	if err != nil {
+		e2elog.Logf("failed to read content from %s %v", rbdDirPath+rbdProvisionerRBAC, err)
+	}
+	_, err = framework.RunKubectlInput(data, "--ignore-not-found=true", ns, "delete", "-f", "-")
+	if err != nil {
+		e2elog.Logf("failed to delete provisioner rbac %s %v", rbdDirPath+rbdProvisionerRBAC, err)
+	}
+
+	data, err = replaceNamespaceInTemplate(rbdDirPath + rbdNodePluginRBAC)
+	if err != nil {
+		e2elog.Logf("failed to read content from %s %v", rbdDirPath+rbdNodePluginRBAC, err)
+	}
+	_, err = framework.RunKubectlInput(data, "delete", "--ignore-not-found=true", ns, "-f", "-")
+	if err != nil {
+		e2elog.Logf("failed to delete nodeplugin rbac %s %v", rbdDirPath+rbdNodePluginRBAC, err)
+	}
+
+	createORDeleteRbdResouces("create")
 }
 
 func deleteRBDPlugin() {
-	_, err := framework.RunKubectl("delete", "-f", rbdDirPath+rbdProvisioner)
+	createORDeleteRbdResouces("delete")
+}
+
+func createORDeleteRbdResouces(action string) {
```

Comment:
somehow I feel above func name is not great :) , we could think some more better name if possible

Reply:
can you please suggest a new name for it?